src/inventorymanager.cpp

```diff
@@ -842,7 +842,8 @@ void ICraftAction::apply(InventoryManager *mgr,
 			count_remaining--;
 
 		// Get next crafting result
-		found = getCraftingResult(inv_craft, crafted, temp, false, gamedef);
+		if (!getCraftingResult(inv_craft, crafted, temp, false, gamedef))
+			break;
```

Comment:
I'm not actually sure about this one, but it seemed like the intention was to break the loop since `found` was set to it's return value and `!found` breaks the loop.